### PR TITLE
Force application update feature

### PR DESCRIPTION
**Pull request by LendiHop**
task: https://www.notion.so/madfissolutions/Development-05abe2b65cd14ae39a101533bc01b708?p=bade699256ba4aa596076ee962a293b4

back: https://github.com/madfish-solutions/templewallet-backend/pull/19

### CODE REVIEW

File: src/hooks/use-timer-effect.hook.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import { DependencyList, useEffect } from 'react';
+
+import { EmptyFn } from '../config/general';
+import { useIsAuthorisedSelector } from '../store/wallet/wallet-selectors';
+
+const effect = (callback: EmptyFn, refreshInterval: number) => {
+  callback();
+
+  let timeoutId = setTimeout(function updateData() {
```
Comment (letier3110) [inline src/hooks/use-timer-effect.hook.ts:9]: why `let` and not `const` ?

File: src/hooks/use-timer-effect.hook.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  return () => clearTimeout(timeoutId);
+};
+
+export const useTimerEffect = (callback: EmptyFn, refreshInterval: number, deps: DependencyList = []) =>
+  useEffect(() => effect(callback, refreshInterval), [...deps]);
+
+export const useAuthorisedTimerEffect = (callback: EmptyFn, refreshInterval: number, deps: DependencyList = []) => {
+  const isAuthorised = useIsAuthorisedSelector();
+
+  useEffect(() => {
```
Comment (letier3110) [inline src/hooks/use-timer-effect.hook.ts:23]: short return `useEffect(() => void (isAuthorised &&  effect(callback, refreshInterval));`

File: src/navigator/main-stack.tsx [hunk 3]
```diff
@@ -63,6 +65,12 @@ export const MainStackScreen = () => {
   useAppLockTimer();
   useBeaconHandler();
 
+  const initAppCheck = () => {
```
Comment (letier3110) [inline src/navigator/main-stack.tsx:68]: short return `const initAppCheck = () => void dispatch(checkApp.submit());`

File: src/navigator/root-stack.tsx [hunk 4]
```diff
@@ -54,6 +56,7 @@ export const RootStackScreen = () => {
 
   useQuickActions();
   const isPasscode = useDevicePasscode();
+  const isForceUpdateNeeded = useCheckInfoSelector().isForceUpdateNeeded;
```
Comment (letier3110) [inline src/navigator/root-stack.tsx:59]: destructurize `const { isForceUpdateNeeded } = useCheckInfoSelector();`

File: src/hooks/use-timer-effect.hook.ts (hunk 1 above)
Comment (LendiHop) [reply src/hooks/use-timer-effect.hook.ts:9]: image

**Review by letier3110 — APPROVED** [review]

File: src/store/app-check/app-check-state.ts [hunk 5]
```diff
@@ -0,0 +1,15 @@
+import { AppCheckInterface } from '../../interfaces/app-check.interface';
+import { createEntity } from '../create-entity';
+import { LoadableEntityState } from '../types';
+
+export interface AppCheckState {
```
Comment (unixvb) [inline src/store/app-check/app-check-state.ts:5]: merge this into `Securty` state

File: src/store/app-check/app-check-state.ts [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+  checkInfo: LoadableEntityState<AppCheckInterface>;
+}
+
+export const appCheckInitialState: AppCheckState = {
+  checkInfo: createEntity({ isForceUpdateNeeded: false })
```
Comment (unixvb) [inline src/store/app-check/app-check-state.ts:10]: mb lets omit `AppCheckInterface` and use `state.isForceUpdateNeeded` ?

**Review by unixvb — APPROVED** [review]

**Pull request merged**